src/vs/editor/contrib/inlineCompletions/browser/inlineCompletionsModel.ts

```diff
@@ -433,6 +435,102 @@ export class InlineCompletionsModel extends Disposable {
 		}
 	}
 
+	private _getEdits(editor: ICodeEditor, completion: SingleTextEdit): { edits: SingleTextEdit[]; editorSelections: Selection[] } {
+
+		const selections = editor.getSelections() ?? [];
+		const secondaryPositions = selections.slice(1).map(selection => selection.getPosition());
+		const primaryPosition = selections[0].getPosition();
+		const textModel = editor.getModel()!;
+		const replacedTextAfterPrimaryCursor = textModel
+			.getLineContent(primaryPosition.lineNumber)
+			.substring(primaryPosition.column - 1, completion.range.endColumn - 1);
+		const secondaryEditText = completion.text.substring(primaryPosition.column - completion.range.startColumn);
+		const edits = [
+			new SingleTextEdit(completion.range, completion.text),
+			...secondaryPositions.map(pos => {
+				const textAfterSecondaryCursor = this.textModel
+					.getLineContent(pos.lineNumber)
+					.substring(pos.column - 1);
+				const l = commonPrefixLength(replacedTextAfterPrimaryCursor, textAfterSecondaryCursor);
+				const range = Range.fromPositions(pos, pos.delta(0, l));
+				return new SingleTextEdit(range, secondaryEditText);
+			})
+		];
+		const newRanges = this._getNewRanges(edits);
+		const editorSelections = newRanges.map(range => Selection.fromPositions(range.getEndPosition()));
+
+		return {
+			edits,
+			editorSelections
+		};
+	}
+
+	// todo: extract this function into a separate place, write some tests for this function
+	// todo: investigate that bug, that we previously saw inside of test 21, there appears to be a logical error somewhere that needs to be resolved
+	private _getNewRanges(edits: SingleTextEdit[]): Range[] {
```

Comment:
This function is not good and maybe buggy.
You can get inspiration from this: https://github.com/microsoft/vscode-copilot/blob/main/src/base/conversation/utils/edit.ts#L108

Also, you can just use `edits.sort(...)` to sort them (or you assume that they are sorted and check the callers).

Then you keep an offset (you can use `Position` for that) and basically use the same algorithm as linked.